gdpr/models.py

```diff
@@ -11,7 +12,64 @@
 
 class LegalReasonManager(models.Manager):
 
-    def create_from_purpose_slug(self, purpose_slug, source_object, issued_at=None, tag=None, related_objects=None):
+    def create_consent(self, purpose_slug, source_object, issued_at=None, tag=None, related_objects=None):
+        """
+        Create (or update, if it exist) a LegalReason with purpose slug for concrete object instance
+
+        Args:
+            purpose_slug: String of Legal Reason purpose
+            source_object: Source object this Legal Reason is related to
+            issued_at: When the Legal Reason consent was given
+            tag:
+            related_objects: Objects this Legal Reason relates to (ie. order, registrations etc.)
+
+        Returns:
+            Legal Reason: LegalReason object
+        """
+        try:
```

Comment:
@matllubos tady jsem předělal ten if/else - lepší?

Reply:
stále je tam problém atomicity

Reply:
jinak použij get_or_create a když ti ho to vrátí tak mu uprav co je potřeba